core/src/main/java/io/questdb/griffin/engine/functions/regex/AbstractLikeVarcharFunctionFactory.java

```diff
@@ -144,13 +148,89 @@ public Function newInstance(
 
     protected abstract boolean isCaseInsensitive();
 
+    /**
+     * Optimized variant of {@link ConstContainsVarcharFunction} with SWAR-based contains implementation.
+     * Works only for patterns up to 8 bytes in size.
+     */
+    private static class ConstContainsSwarVarcharFunction extends BooleanFunction implements UnaryFunction {
+        private static final int MAX_SIZE = Long.BYTES;
+        private final Utf8Sequence pattern; // only used in toPlan
+        private final long patternMask;
+        private final int patternSize;
+        private final long patternWord;
+        private final Function value;
+
+        public ConstContainsSwarVarcharFunction(Function value, Utf8Sequence pattern) {
+            assert pattern.size() > 0 && pattern.size() <= MAX_SIZE;
+            this.value = value;
+            this.patternSize = pattern.size();
+            this.patternMask = patternSize == MAX_SIZE ? -1L : (1L << 8 * pattern.size()) - 1L;
+            long patternWord = 0;
+            for (int i = 0, n = pattern.size(); i < n; i++) {
+                patternWord |= (long) (pattern.byteAt(i) & 0xff) << (8 * i);
+            }
+            this.patternWord = patternWord;
+            this.pattern = pattern;
+        }
+
+        @Override
+        public Function getArg() {
+            return value;
+        }
+
+        @Override
+        public boolean getBool(Record rec) {
+            Utf8Sequence us = value.getVarcharA(rec);
+            if (us == null || us.size() < patternSize) {
+                return false;
+            }
+
+            int size = us.size();
+
+            int i = 0;
+            for (int n = size - MAX_SIZE + 1; i < n; i++) {
+                if ((us.longAt(i) & patternMask) == patternWord) {
+                    return true;
+                }
```

Comment:
Another optimization perhaps worth trying: Use SWAR to locate the first occurrence of the first byte from the constant pattern. 

Chances are the loaded `long` won't have the byte at all. If that's the case then you save yourself 8 long comparisons. Even if there is the first byte in the `long` you might save yourself some comparisons - since it allows you to 'fast-forward' right into the offset where the pattern may possibly start.

Reply:
That's a good idea. Let me try it.

Reply:
Here it goes: 6c28d86d6d9cfcc03f8880c61198ac4d395fe613

I've also updated query exec times in the PR description.

Reply:
Since this code doesn't do anything with the result of `checkZeroByte` except compare it with zero, it seems this call can be removed and the result of XOR can be checked against 0 directly. Alternatively, and to better protect against worst-case patterns, we can check only the specific offsets where the byte was found.

Reply:
epic!